### PR TITLE
[libhsplasma] Add new port

### PR DESCRIPTION
**Pull request by Hoikas**
This adds a port for [libHSPlasma](https://github.com/H-uru/libhsplasma).

### CODE REVIEW

**Comment by Hoikas** [comment]
I'm converting this to a draft in order to deal with CI failures and an issue with `HSPlasma_INCLUDE_DIRS`, which may require an update to the upstream library.

**Comment by JackBoosY** [comment]
Failure logs [here](https://dev.azure.com/vcpkg/public/_build/results?buildId=42100&view=artifacts&type=publishedArtifacts).

**Comment by Hoikas** [comment]
Blocking on upstream H-uru/libHSPlasma#184, which should fix the CI issues.

**Comment by Hoikas** [comment]
Thanks for your patience @JackBoosY. This is now ready for your review.

File: ports/libhsplasma/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO H-uru/libhsplasma
+    REF afd9c46fa72afbbaf5260800ecfde0704ba2a475
+    SHA512 21774b465ede7fe5e44df4470ed6867977ac9249400ba6d0bed33fd0e856d467de1afe79afb8a4dc6145eef21c5a3f79590b5cfa3ea1337cc355ca8f3bf1a11c
+    HEAD_REF master
+)
+
+if(net IN_LIST FEATURES)
+    set(HSPLASMA_NET ON)
+endif()
```
Comment (JackBoosY) [inline ports/libhsplasma/portfile.cmake:13]:
```suggestion
vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
    net HSPLASMA_NET
)
```

File: ports/libhsplasma/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
+    OPTIONS
```
Comment (JackBoosY) [inline ports/libhsplasma/portfile.cmake:18]:
```suggestion
    OPTIONS
        ${FEATURE_OPTIONS}
```

File: ports/libhsplasma/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    CONFIG_PATH share/cmake/HSPlasma
+    TARGET_PATH share/HSPlasma
+)
+vcpkg_fixup_pkgconfig()
+vcpkg_copy_pdbs()
+
+vcpkg_test_cmake(HSPlasma)
+if(HSPLASMA_NET)
+    vcpkg_test_cmake(HSPlasmaNet)
+endif()
```
Comment (JackBoosY) [inline ports/libhsplasma/portfile.cmake:36]:
```suggestion
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
Test feautres successful on `x86-windows` and `x64-windows`.

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**